src/validation/async/index.ts

```diff
@@ -0,0 +1,260 @@
+import { Request } from 'express';
+import { ValidationError } from "express-validator";
+
+import * as config from '../../config';
+import { ApplicationData } from "model";
+import { RoleWithinTrustType } from '../../model/role.within.trust.type.model';
+import { ErrorMessages } from '../../validation/error.messages';
+import { checkCeasedDateOnOrAfterDateOfBirth, checkCeasedDateOnOrAfterTrustCreationDate, checkIndividualCeasedDateOnOrAfterInterestedPersonStartDate, checkLegalEntityCeasedDateOnOrAfterInterestedPersonStartDate, historicalBOEndDateContext, historicalBOStartDateContext } from '../../validation/fields/date.validation';
+import { hasNoBoAssignableToTrust } from '../../utils/trusts';
+import { dateContext } from '../../validation/fields/helper/date.validation.helper';
+import { checkDatePreviousToFilingDate } from '../../validation/custom.validation';
+
+export const checkTrustStillInvolved = (appData: ApplicationData, req): ValidationError[] => {
```

Comment:
A comment (somewhere) explaining why these functions are sitting in a folder named `async` would be helpful. I know this is probably related to the async call to get application data but saying that `appData` is being passed in to these functions so why are they asynchronous?

Reply:
They need to make a call to getApplication, which is why they have been moved
But as we already make the getApplication call at controller level, I pass appData to avoid making two (or more) calls

Reply:
or I could put them in a separate file
because they are not async but require data to be fetched asynchronously

Reply:
A separate file might be better, as I was expecting this file to contain only async functions.

Reply:
stillInvolved validators have been moved to a separate file